src/combined.cc

```diff
@@ -0,0 +1,217 @@
+    Ipv4InterfaceContainer interfaces_1_2 = address.Assign(devices_1_2);
+    address.NewNetwork();
+    address.SetBase("10.1.2.0", "255.255.255.0");
+    Ipv4InterfaceContainer interfaces_2_3 = address.Assign(devices_2_3);
+    address.NewNetwork();
+    address.SetBase("10.1.3.0", "255.255.255.0");
+    Ipv4InterfaceContainer interfaces_2_4 = address.Assign(devices_2_4);
+    address.NewNetwork();
+    address.SetBase("10.1.4.0", "255.255.255.0");
+    Ipv4InterfaceContainer interfaces_4_3 = address.Assign(devices_4_3);
+    address.NewNetwork();
+    address.SetBase("10.1.5.0", "255.255.255.0");
+    Ipv4InterfaceContainer interfaces_3_5 = address.Assign(devices_3_5);
+    address.NewNetwork();
+    Ipv4GlobalRoutingHelper::PopulateRoutingTables();
+
+    // Receiver address
+    Ipv4Address receiver_addr = interfaces_3_5.GetAddress(1);
+
+    // UDP Congestion traffic setup
+    uint16_t udp_port = 5001;
+    OnOffHelper udp_source("ns3::UdpSocketFactory",
+                           InetSocketAddress(receiver_addr, udp_port));
+    udp_source.SetAttribute(
+        "OnTime", StringValue("ns3::ConstantRandomVariable[Constant=1]"));
+    udp_source.SetAttribute(
+        "OffTime", StringValue("ns3::ConstantRandomVariable[Constant=0]"));
+    udp_source.SetAttribute("DataRate", DataRateValue(DataRate("1Mbps")));
+    udp_source.SetAttribute("PacketSize", UintegerValue(1024));
+
+    ApplicationContainer udp_app = udp_source.Install(nodes.Get(0));
+    udp_app.Start(Seconds(2.0));
+    udp_app.Stop(Seconds(6.0));
+
+    // TCP Setup
+    SetupTCPConfig();
+    uint16_t tcp_port = 5002;
+    BulkSendHelper tcp_source("ns3::TcpSocketFactory",
+                              InetSocketAddress(receiver_addr, tcp_port));
+    tcp_source.SetAttribute("MaxBytes", UintegerValue(1000)); // Tweak this
+    ApplicationContainer tcp_app = tcp_source.Install(nodes.Get(1));
+    tcp_app.Start(Seconds(0.0));
+    tcp_app.Stop(Seconds(10.0));
+
+    // Packet sink setup (Receiver node)
+    PacketSinkHelper sink("ns3::TcpSocketFactory",
+                          InetSocketAddress(Ipv4Address::GetAny(), tcp_port));
+    ApplicationContainer sink_app = sink.Install(nodes.Get(5));
+    tcp_app.Start(Seconds(0.0));
+    tcp_app.Start(Seconds(10.0));
+
+    // NOTE: Is TrafficControlHelper needed here?
+
+    // LFA Alternate Path setup
+    // Set up an alternate forwarding target, assuming you have an alternate
+    // path configured
+
+    // TODO: Need some help with setting alternate target
+    // setAlternateTarget<0>(devices01, getDevice<0>(devices02));
+    // setAlternateTarget<0>(devices02, getDevice<0>(devices01));
+
+    // setAlternateTarget<0>(devices12, getDevice<1>(devices01));
+    // setAlternateTarget<1>(devices01, getDevice<0>(devices12));
+
+    // setAlternateTarget<1>(devices02, getDevice<1>(devices12));
+    // setAlternateTarget<1>(devices12, getDevice<1>(devices02));
```

Comment:
this needs to be linked to the topology, and thats why the lfa policy is giving a seg fault. I will change it on my part to print but not abort the error